app/utils/vcs.js

```diff
@@ -0,0 +1,59 @@
+import config from 'travis/config/environment';
+
+const { providers } = config;
+
+const vcsId = (vcsType) => {
+  const availableProvidersRegex = Object.keys(providers).join('|');
+  const match = (vcsType || 'github').match(new RegExp(availableProvidersRegex, 'i'));
+
+  if (!match[0]) {
+    throw new Error(`Invalid VCS Type "${vcsType}"`);
+  }
+  return match[0].toLowerCase();
+};
+
+const vcsConfig = (vcsType) => (
+  providers[vcsId(vcsType)]
+);
+
+const replaceParams = (template, params) => (
+  Object
+    .keys(params)
+    .reduce((url, key) => url.replace(`:${key}`, params[key]), template)
+);
+
+const templateParams = (template) => (
+  template
+    .match(/:[a-z]+/g)
+    .map((param) => param.slice(1))
+);
+
+const paramsValid = (template, params) => (
+  arrayEqual(Object.keys(params), templateParams(template))
+);
+
+const arrayEqual = (array1, array2) => (
+  array1.sort().toString() == array2.sort().toString()
+);
+
+export const vcsUrl = (resource, vcsType, params) => {
```

Comment:
I'd suggest to set a default value for `params = {}`, because it's public and if it won't be set from the outside, many methods here (that use `Object.keys`) will break.